src/adapter/etl-adapter-filesystem/src/Flow/ETL/Adapter/Filesystem/FlysystemFS.php

```diff
@@ -132,6 +132,10 @@ public function rm(Path $path) : void
      */
     public function scan(Path $path, PartitionFilter $partitionFilter = new NoopFilter()) : \Generator
     {
+        if (!$path->isPattern() && !$this->fileExists($path) && !$this->directoryExists($path)) {
```

Comment:
I'm not sure if we should keep that `!$this->directoryExists($path)` since it's creating that weird behavior that I mentioned before. 

`/path/to/dir` will scan the entire dir, just like `/path/to/dir/**/*` 

And I think that since we support glob, auto scanning entire directory by `scan` method might not be necessary.

Reply:
hey @rzarno 
Are there any issues with removing that extra condition?

Reply:
hey Norbert,
yes, I originally made it without directory related condition as you suggested  and some additional unit tests broke  
so I need to take a deeper look, but didn't had time yet

Reply:
sounds good, I just wanted to check if your aren't stuck somewhere with this. Those tests should be pretty straightforward to fix as you just need to adjust the paths from `/path/to/folder` into something like this `/path/to/folder/**/*.json`.
But if there would be any edge cases that I'm missing now, please let me know

Reply:
yes, indeed, wildcards were enough to solve issues with units
updated PR, thanks for advice